File: app/models/competitor.rb
class Competitor < ActiveRecord::Base
	validates :name, length:{ maximum: 64 }, presence: true
	validates :group_num, :wins, :loss, :draw, :placement_num, numericality:{ only_integer:true }, allow_blank: true
	
	has_many :matches, through: :competitor_matches
	belongs_to :tourney, inverse_of: :competitors
end

File: app/main/lib/league/competitor.ts
class Competitor {
  public id: number;
  public name: string;
  public tier: number;

  constructor( id: number, name: string, tier: number = null ) {
    this.id = id;
    this.name = name;
    this.tier = tier;
  }
}


export default Competitor;
